sympy/solvers/ode/ode.py

```diff
@@ -7631,8 +7631,11 @@ def _linear_2eq_order1_type4(x, y, t, r, eq):
         G = Integral(r['b'], t)
         sol1 = F*(C1*cos(G) + C2*sin(G))
         sol2 = F*(-C1*sin(G) + C2*cos(G))
+    # FIXME: the case below doesn't seem correct, is only XFAIL tested and doesn't
+    # match the description in the docstring above. It can be triggered with:
+    # dsolve([Eq(f(x).diff(x), f(x) + x*g(x)), Eq(g(x).diff(x), x*f(x) - g(x))])
```

Comment:
We know the `r['d'] == -r['a']` case is wrong so let's raise a NotImplemented rather than returning a wrong solution (or modifying it to return a different wrong solution):
```python
eqs = [
Eq(f(x).diff(x), f(x) + y*g(x)), 
Eq(g(x).diff(x), f(x)*y - g(x))]
ans=dsolve(eqs)
[i.subs([i.args for i in ans]).doit().expand()==0 for i in eqs] # -> [False, False]
```
The math of the docstring could also be changed to reflect that `F = e^{...}`.

Reply:
I think it would be better just to delete this particular solver. The code is broken and it is a special case of something more general. We can have a system of ODEs:
```
X(t).diff(t) = A(t)*X(t)
```
where `X` is a column matrix and `A` is a square matrix. If the matrix `A(t)` is symmetric and its integral is `B(t)` (also symmetric) then the solution is `X(t) = exp(B(t))`. This makes a solver that can work for any number of equations rather than just two. The condition can be further generalised by saying that we require only that `A` and `B` commute (they don't need to be symmetric necessarily).

Reply:
OK, so let's get these changes in. I will open an issue for this case.